### ISSUE TITLE
Query for nested field in 2nd level failing

### ISSUE THREAD

**Issue by xavimartimrf** [ppadovani/KibanaNestedSupportPlugin#71]
**Plugin version**: 1.0.2

**Kibana version**: 6.2.4

**Plugins installed**: Only nested-fields-support

**Description of the problem including expected versus actual behavior**:
I have tried this plugin in the dummy indexes available [here](https://www.elastic.co/guide/en/kibana/current/tutorial-load-dataset.html), and when querying through the Search bar in the Discover application it seems to work fine, returning every document that matches the condition.

However, when trying in my real index, I am using a nested field "y" in a 2nd level, meaning the name of the field is "x.y" and its attributes are "x.y.z". "x" is not an array, just a field group.

Then, when querying: ' x.y.z = "something" '
It returns me 0 results, even if there are several documents that match the condition.
When querying ' NOT x.y.z = "something" ' it returns me all the documents.

Is it possible that the plugin does not support nested fields in 2nd level?

**Steps to reproduce**:
 1. Create nested field in 2nd level.
 2. Try querying a certain attribute of the nested field.

**Comment by ppadovani** [comment]
It shouldn't matter how many levels deep the nesting is.. I'll take a look.

**Comment by ppadovani** [comment]
Ok... within my standard test index (see the /scripts directory of this repo), I added 'options' array to the cars array a house can have. I am able to query against the options array for specific options like:

cars.options.name="V6" and get back proper matches.

You can test against my test data set by pulling/downloading the scripts directory, and loading the test data into elastic search. The scripts generates random data against a known elastic search mapping.

What DOESN'T work is the result summary. It doesn't show the next level sub-list of options. I will work on a fix for that as part of this issue.

**Comment by xavimartimrf** [comment]
Ok, then it's probably not related to 2nd level. I have generated the homes Index with 100 homes, and:
- It does not work when querying exact match, but it does with LIKE (?):
![image](https://user-images.githubusercontent.com/28809974/39928637-35fd4fee-5536-11e8-99af-be5d3ea305df.png)
![image](https://user-images.githubusercontent.com/28809974/39928642-38d7cf3c-5536-11e8-8d03-09b7af12bee2.png)
![image](https://user-images.githubusercontent.com/28809974/39928649-3ca2c95a-5536-11e8-8df3-17c830b8991a.png)
![image](https://user-images.githubusercontent.com/28809974/39928662-3ffe1528-5536-11e8-840a-55b9bcc082c2.png)

In my index, it does not even work with LIKE yet. I cannot figure it out.

**Comment by ppadovani** [comment]
Did you create the index with the home.json file that is in the scripts directory? It has the proper mapping created. I'm seen the above behavior if the mapping isn't created and Elastic does a dynamic mapping. You can see if this is the case by pulling the mapping from your created index and compare it to the home.json file.

**Comment by ppadovani** [comment]
I've got the summary fixed... and am working on the individual row summary for sub-nested fields.

**Comment by ppadovani** [comment]
Fixed in all 6.2.X releases.

**Comment by xavimartimrf** [comment]
You are right, looks like index mapping was created dynamically, and therefore no field was defined as "nested". I don't know why, I just ran your script.
I have created now a template for homes*, with home.json file, then I deleted the index and ran genData again. Index has the correct mapping now.

However, I keep having the same problem or even worse:
![image](https://user-images.githubusercontent.com/28809974/39993138-287fd418-5775-11e8-99eb-b1a2c66155fa.png)

![image](https://user-images.githubusercontent.com/28809974/39993155-3482c766-5775-11e8-8fba-dfd8f6c3bc63.png)

As now LIKE won't work either:
![image](https://user-images.githubusercontent.com/28809974/39993171-3f4c0de2-5775-11e8-825e-5fd14abc6ee3.png)

**Comment by ppadovani** [comment]
Can you try and put the mapping in as an index mapping and not a template? I use this command:

curl -X PUT -H "Content-Type: application/json" localhost:9200/homes -d @home.json

**Comment by xavimartimrf** [comment]
Deleted template & index, used your command, reran script. Same result for now. Attaching the JSON resulting of calling http://localhost:9200/homes, but I think it is the same as your home.json.
[homes.txt](https://github.com/ppadovani/KibanaNestedSupportPlugin/files/2000941/homes.txt)

**Comment by ppadovani** [comment]
So... The only way I can duplicate this is if I turn off the nested configuration for the homes index. Can you make sure that you have that turned on? Management->Nested Fields. Should like this if it is on:

<img width="1675" alt="screen shot 2018-05-14 at 8 56 18 am" src="https://user-images.githubusercontent.com/11234684/40001832-b5c2d1d4-5754-11e8-9cd8-6ec72dfbaefd.png">

Notice this 'check' under actions...

**Comment by xavimartimrf** [comment]
Got it. I had it checked, but I tried unchecking and rechecking and now it is working fine.

My explanation:
- Somehow running your script created a dynamic default index mapping without "nested" types.
- When checking the box, the plugin did not detect any nested field and therefore did not act upon them.
- I fixed the mapping, but the plugin did not look again into that.

Maybe you can fix it so it checks the mapping every time it is updated, or inform about the need to recheck the box in that case. Anyways, thanks a lot :)

**Comment by ppadovani** [comment]
Great! I'm glad you got that working!

There are a couple of things I want to fix in relation to that configuration screen and the refresh, but fixing active query bugs takes precedence right now. I'm actively choosing the next major version feature list for this plugin, and fixing that will be near the top.

**Comment by ppadovani** [comment]
Ooops.. I need to keep this open until I've finished the back porting of the fixes I did make for this issue.

**Comment by ppadovani** [comment]
Fixed in all 6.X releases

**Comment by ppadovani** [comment]
Fixed in 5.6.9
